File: www.app.com/core-api/src/main/java/com/base/coreapi/model/calendar/Reservation.java
package com.base.coreapi.model.calendar;

import com.base.coreapi.model.auth.ApplicationUser;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Data
@Entity
@NoArgsConstructor
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    @ManyToOne
    private ApplicationUser user;

    @JsonIgnore
    @ManyToOne
    private Event event;
}

File: www.app.com/frontend-app/src/app/model/backend/calendar/reservation.ts
import {ApplicationUser} from "../auth/application-user";

export class Reservation {

  public id: number;
  public user: ApplicationUser;
  public event: string;
  public date: Date;
  public childName: string;

  constructor(event, date, childName) {
    this.event = event;
    this.date = date;
    this.childName = childName;
  }
}
